src/components/Main/Main.tsx

```diff
@@ -0,0 +1,126 @@
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { useEffect, useRef } from 'react';
+
+interface Props {
+  todos: Todo[];
+  loader: boolean;
+  tempTodo: Todo | null;
+  deletePosts: (b: Todo) => void;
+  addActivePosts: (b: Todo) => void;
+  loadTodos: number[];
+  changeInputs: (b: Todo, c: string) => void;
+  clickTodo: number | null;
+  setChangeInput: (b: string) => void;
+  changeInput: string;
+  setClickTodo: (b: number) => void;
+}
+
+export const Main = ({
+  todos,
+  loader,
+  tempTodo,
+  deletePosts,
+  addActivePosts,
+  loadTodos,
+  changeInputs,
+  clickTodo,
+  setChangeInput,
+  changeInput,
+  setClickTodo,
+}: Props) => {
+  const inputFocus = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (clickTodo !== null && inputFocus.current) {
+      inputFocus.current.focus();
+    }
+  }, [clickTodo]);
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.length > 0 &&
+        todos.map(todo => (
+          <div
+            data-cy="Todo"
+            className={cn('todo', { completed: todo.completed })}
+            key={todo.id}
+          >
+            <label
+              className="todo__status-label"
+              aria-labelledby={`todo_${todo.id}`}
+            >
+              <input
+                id={`todo_${todo.id}`}
+                data-cy="TodoStatus"
+                type="checkbox"
+                className="todo__status"
+                onClick={() => {
+                  addActivePosts(todo);
+                }}
+              />
+            </label>
+
+            {todo.id === clickTodo ? (
+              <form>
+                <input
```

Comment:
you can rewrite your code to render TodoItem here in the map

Reply:
Still not fixed 
<img width="913" alt="image" src="https://github.com/mate-academy/react_todo-app-with-api/assets/64750958/8302aada-8a05-47d4-b1f2-0464eb3cd712">